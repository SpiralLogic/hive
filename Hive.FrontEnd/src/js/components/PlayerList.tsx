import * as React from 'react';
import { useHiveContext } from '../game-context';
import PlayerTiles from './PlayerTiles';

function PlayerList() {
    const { players } = useHiveContext();
    return (
        <div className="players">
            {players.map((player) => (
                <PlayerTiles key={player.id} {...player} />
            ))}
        </div>
    );
}

PlayerList.displayName = 'Player List';

export default PlayerList;
