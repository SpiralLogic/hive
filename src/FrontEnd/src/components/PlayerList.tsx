import { FunctionComponent, h } from 'preact';
import { PlayerId, Players } from '../domain';
import ActiveTile from './ActiveTile';
import Player from './Player';

const Players: FunctionComponent<{ players: Players; currentPlayer: PlayerId }> = (props) => {
  const { players, currentPlayer } = props;
  return (
    <aside className="players">
      {players.map((player) => (
        <Player name={player.name} id={player.id} currentPlayer={currentPlayer} hide={!!player.tiles.length}>
          {player.tiles.map((tile) => (
            <ActiveTile key={tile.id} {...tile} />
          ))}
        </Player>
      ))}
    </aside>
  );
};

Players.displayName = 'Player List';
export default Players;
