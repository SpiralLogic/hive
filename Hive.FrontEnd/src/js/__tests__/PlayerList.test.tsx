import { render } from '@testing-library/preact';
import { h } from 'preact';
import PlayerList from '../components/PlayerList';

const ant = { id: 1, playerId: 1, content: 'ant', availableMoves: [{ q: 1, r: 1 }] };
const fly = { id: 2, playerId: 0, content: 'fly', availableMoves: [] };

const players = [
    { id: 1, name: 'Player 1', availableTiles: [ant, fly, fly] },
    { id: 2, name: 'Player 2', availableTiles: [ant] }
];

const props = {players: players };
let playerList: HTMLElement;

beforeEach(() => {
    playerList = render(<PlayerList {...props}/>).container.firstElementChild as HTMLElement;
});

describe('Player List', () => {
    test('to have class', () => {
        expect(playerList).toHaveClass('players');
    });

    test('players are rendered', () => {
        expect(playerList.getElementsByClassName('player')).toHaveLength(2);
    });
});

describe('snapshot', () => {
    test('matches current snapshot', () => {
        expect(playerList).toMatchSnapshot();
    });
});
