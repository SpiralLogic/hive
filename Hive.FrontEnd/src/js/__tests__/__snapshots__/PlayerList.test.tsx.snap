// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches current snapshot 1`] = `
HTMLCollection [
  <div
    class="players"
  >
    <div
      class="player"
      title="Player 1"
    >
      <span
        class="name"
      >
        Player 1
      </span>
      <div
        class="tiles"
      >
        <div
          class="hex tile"
          draggable="true"
          style="--color: #f64c72;"
          title="ant"
        >
          <span>
            ant
          </span>
        </div>
        <div
          class="hex tile"
          draggable="false"
          style="--color: #85dcbc;"
          title="fly"
        >
          <span>
            fly
          </span>
        </div>
        <div
          class="hex tile"
          draggable="false"
          style="--color: #85dcbc;"
          title="fly"
        >
          <span>
            fly
          </span>
        </div>
      </div>
    </div>
    <div
      class="player"
      title="Player 2"
    >
      <span
        class="name"
      >
        Player 2
      </span>
      <div
        class="tiles"
      >
        <div
          class="hex tile"
          draggable="true"
          style="--color: #f64c72;"
          title="ant"
        >
          <span>
            ant
          </span>
        </div>
      </div>
    </div>
  </div>,
]
`;
